components/tools/OmeroWeb/omeroweb/webgateway/templates/webgateway/viewport/omero_image.html

```diff
@@ -1182,6 +1197,40 @@ <h3 class="can-collapse" id="bulk-annotations" style="display: none;">Bulk Annot
         }
     }
 
+    var check_visibility_column = function() {
+        var roi_checkboxes = $('.roi_visibility');
+        var shape_checkboxes = $('.shape_visibility');
+        for (x=0; x<roi_checkboxes.length; x++) {
+            if (roi_checkboxes[x].checked) {
+                $('#toggle_roi_visibility').prop('checked', true);
+                return;
+            }
+        }
+        for (x=0; x<shape_checkboxes.length; x++) {
+            if (shape_checkboxes[x].checked) {
+                $('#toggle_roi_visibility').prop('checked', true);
+                return;
+            }
+        }
```

Comment:
Just a suggestion: I think you can use 
```
$('.roi_visibility').is(":checked")
$('#toggle_roi_visibility').is(":checked")
```
instead of the loops you have to check if any of those checkboxes are checked.